app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt

```diff
@@ -1,62 +1,84 @@
 package com.pvp.app.ui.screen.profile
 
+import androidx.compose.ui.graphics.ImageBitmap
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.pvp.app.api.AuthenticationService
 import com.pvp.app.api.UserService
+import com.pvp.app.model.Ingredient
 import com.pvp.app.model.SignOutResult
+import com.pvp.app.model.SportActivity
 import com.pvp.app.model.User
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+
 @HiltViewModel
 class ProfileViewModel @Inject constructor(
     private val authenticationService: AuthenticationService,
     private val userService: UserService
 ) : ViewModel() {
 
-    private val _user = MutableStateFlow<User?>(null)
-    val user = _user.asStateFlow()
+    private val _state = MutableStateFlow(
+        ProfileState(
+            isLoading = true,
+            user = User(),
+            userAvatar = ImageBitmap(1, 1)
+        )
+    )
+    val state: StateFlow<ProfileState> = _state.asStateFlow()
 
     init {
         viewModelScope.launch {
-            _user.value = userService
+            val user = userService
```

Comment:
This makes user be initialized only once. No outer changes will be noticed. Try to take the flow, transform it and launch it into the background `launchIn(scope)`.